main.cpp

```diff
@@ -250,7 +487,10 @@ struct ping_queue_t
 
 		*out = m_queue.front();
 		m_queue.pop_front();
-		m_ips.erase(out->ep);
+		if (out->ep.protocol() == udp::v4())
```

Comment:
sorry, I mixed this up with the peer_buffer_t. That's presumably where the bulk of the storage is used, and that *is* a template.